src/pages/listing/ListingGallery.tsx

```diff
@@ -1,20 +1,89 @@
 import * as React from 'react';
-import { Button, Row } from 'reactstrap';
+import { Alert, Button, Col, Modal, ModalHeader, ModalBody, Row } from 'reactstrap';
+import CopyToClipboard from 'react-copy-to-clipboard';
+import { withRouter } from 'react-router-dom';
 
 import ImageGrid from 'legacy/shared/ImageGrid';
 import CarouselPortal from 'legacy/work/CarouselPortal';
 
 import { Listing } from 'networking/listings';
 
-const ListingGallery = ({ listingPicUrl, photos }: Listing) => {
+interface Props extends RouterProps, Listing { };
+
+const ListingGallery = (props: Props) => {
   const [isOpen, setOpen] = React.useState<boolean>(false);
+  const [isModalOpen, setModal] = React.useState<boolean>(false);
+  const [isAlertOpen, setAlert] = React.useState<boolean>(false);
+
+  const { listingPicUrl, location, photos, title } = props;
+  const listingLink = `https://beenest.com${location.pathname}`;
   return <Row className="w-100 height-60vh px-0 mx-0 position-relative">
     <ImageGrid images={[listingPicUrl, ...photos]} />
+    <Button className="position-absolute top-0 right-0 m-4" color="primary" onClick={() => toggleModal()} type="button">
+      <span className="far fa-share-square"></span>
+    </Button>
+    <Modal isOpen={isModalOpen} toggle={() => toggleModal()}>
```

Comment:
This looks good! Can we move this out to its own component? Picturing usage like:

    <ShareModal isOpen={...} toggle={...} url={listingLink} title={title} />

With this component inline, most of the lines of code in `ListingGallery` end up being about the sharing button, which is unexpected.

Reply:
makes sense! will move it out to it's own component